src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -27,6 +27,34 @@ private[spark] object Config {
   private val invalidLabelListErrorMessage =
     "Must be a comma-separated list of valid Kubernetes labels (each as key=value)"
 
+  val ARMADA_SERVER_INTERNAL_URL: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.internalUrl")
+      .doc(
+        "The Kubernetes DNS or IP address of the Armada Server. " +
+          "This URL is used by the Driver when running in Cluster mode. " +
+          "If not specified, 'spark.master' will be used."
+      )
+      .stringConf
+      .createWithDefaultString("")
+
+  val ARMADA_JOB_QUEUE: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.queue")
+      .doc(
+        "The name of the job queue to use for the Armada job."
+      )
+      .stringConf
+      .checkValue(_.nonEmpty, "Queue name must be provided.")
+      .createWithDefault("")
+
+  val ARMADA_JOB_SET_ID: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.jobSetId")
+      .doc(
+        "The JobSet ID for which the driver and executor pods will be part of."
+      )
+      .stringConf
+      .checkValue(_.nonEmpty, "Job set ID must be provided.")
+      .createWithDefault("")
```

Comment:
```suggestion
      .stringConf
      .createOptional
```

Reply:
Please consider this note: https://github.com/armadaproject/armada-spark/pull/40/files#r2083053971